app/controllers/concerns/grouping_helper.rb

```diff
@@ -0,0 +1,68 @@
+module GroupingHelper
+
+  class GroupedIncumbency
+    attr_accessor :start_date, :end_date, :nodes, :grouped_by, :type
+  end
+
+  # Returns array of grouped, ungrouped and unknown objects
+  def self.group(data, category, property)
+    grouped_data = group_data(data, category, property)
+    create_sorted_array(grouped_data)
+  end
+
+  # Returns hash with grouped keys
+  # e.g. { 12345678 => [...], 23456789 => [...], "UNKNOWN" => [...] }
+  def self.group_data(data, category, property)
+    data.group_by { |data_point| data_point.try(category).try(property) || 'UNKNOWN' }
+  end
+
+  # Create array with all UNKNOWN nodes and create array with all ungrouped data
+  def self.create_sorted_array(data_hash)
+    sorted_array = []
+    data_hash.each do |key, value|
+      if key == 'UNKNOWN' || value.length == 1
```

Comment:
I'd add a comment here making it clear that you don't need to group a single value - took me a second to work out what the length check meant ><